LiteCore/Support/FilePath.cc

```diff
@@ -80,13 +81,30 @@ static int copyfile(const char* from, const char* to)
         errno = e;
         return write_fd;
     }
-    
-    if(sendfile(write_fd, read_fd, &offset, stat_buf.st_size) < 0) {
-        int e = errno;
-        close(read_fd);
-        close(write_fd);
-        errno = e;
-        return -1;
+
+    size_t expected = stat_buf.st_size;
+    ssize_t bytes = 0;
+    int noProgressRetry = 2;
+    while (bytes < expected) {
+        expected -= bytes;
```

Comment:
There's no reason to do this before the first copy; why not move it to the end of the loop, i.e. after `noProgressRetry = 2;`?

Reply:
This is because I wanted to avoid subtraction of unsigned without comparison, like 
while (expected > 0) {
 bytes = ...
 expected -= bytes;
}
although in theory we should have Assert(bytes <= expected).